engine/storage/src/org/apache/cloudstack/storage/image/BaseImageStoreDriverImpl.java

```diff
@@ -79,6 +80,25 @@
     VMTemplateZoneDao _vmTemplateZoneDao;
     @Inject
     AlertManager _alertMgr;
+    @Inject
+    ImageStoreDetailsDao _imageStoreDetailsDao;
+
+    private static final String NFS_VERSION_DETAILS_KEY = "nfs.version";
```

Comment:
@nvazquez One way would be create an intermediate class for NFS based stores and derive CloudStackImageStoreDriverImpl and SimulatorImageStoreDriverImpl from that.